mlir/lib/Dialect/Tosa/IR/TosaOps.cpp

```diff
@@ -459,16 +472,123 @@ LogicalResult tosa::AvgPool2dOp::verify() {
   if (inputETy.isF32() && !accType.isF32())
     return emitOpError("accumulator type for f32 tensor is not f32");
 
+  if ((llvm::isa<Float8E5M2Type>(inputETy) ||
+       llvm::isa<Float8E4M3FNType>(inputETy)) &&
+      !accType.isF16())
+    return emitOpError("accumulator type for f8 tensor is not f16");
+
   if ((inputETy.isF32() && resultETy.isF32()) ||
```

Comment:
Do we plan to keep growing this? We could probably simplify it a bit

Reply:
I don't see we'll grow this in the short term. It reads clear to me. If we refactor this into another function, it takes additional time for me to find that function and check.

Reply:
removed.